foursight_core/react/api/jwt_utils.py

```diff
@@ -0,0 +1,62 @@
+import copy
+import jwt as jwtlib
+
+
+def jwt_encode(value: dict, audience: str, secret: str) -> str:
+    """
+    JWT signs and encodes the given (dictionary) value and returns its string.
+    Uses the given audience (aka Auth0 client ID) and associated secret for signing.
+    If the given value does NOT already have an "aud" property, then we add it,
+    setting it to the given audience value; or if it is present and a non-empty
+    audience is given then it is used as the "aud" property. If no "aud" property
+    is present and no audience is given, or ie any other error occurs, then an
+    exception will be raised.
+
+    Note that strictly speaking, a JWT may be created (signed/encoded) WITHOUT
+    an "aud" property, but an exception will be raised when attempting
+    to verify/decode such a JWT; so best to disallow that case here.
+    The exception we'd get verifying/decoding such a JWT looks like this FYI:
+    Exception verifying/decoding JWT: Token is missing the "aud" claim
+    """
+    if not value:
+        raise Exception("Attempt to encode JWT for empty value.")
+    elif not isinstance(value, dict):
+        raise Exception("Attempt to encode JWT for non-dictionary value.")
+    if audience:
+        # Note: if an audience (aud) is NOT present in the value to sign/encode then
+        # jwtlib will raise: Exception decoding JWT - Token is missing the "aud" claim.
+        value_aud = value.get("aud")
+        if value_aud != audience:
+            # If given an audience, and it doesn't match the audience in
+            # the given value, then make a copy (as we don't want to change
+            # the given value out from under the caller), and update it.
+            value = copy.deepcopy(value)
+            value["aud"] = audience
+    if not value.get("aud"):
+        raise Exception("Cannot encode object as JWT without an 'aud' property")
+    try:
+        encoded_value = jwtlib.encode(value, secret, algorithm="HS256")
+        if isinstance(encoded_value, bytes):
+            # Depending on version jwtlib.encode returns bytes or string :-(
+            encoded_value = encoded_value.decode("utf-8")
+        return encoded_value
+    except Exception as e:
+        raise Exception(f"Exception signing/encoding JWT: {str(e)}")
```

Comment:
If you're going to just show the message, use `{e}` since `str` is what it will call to display it anyway. You're basically writing `str(str(e))` here, which will work fine but is not really necessary. But I somewhat suggest importing `get_error_message` from `dcicutils.misc_utils` and then doing:
```suggestion
        raise Exception(f"Exception signing/encoding JWT: {get_error_message(e)}")
```
which will arrange to put the error class ahead of the message. Sometimes it's useful to know that part. But either way is fine, depending on your need. The reason I like the error class is that in the case of `KeyError`, if you don't have that, the message will be the key name _only_ and that is sometimes completely baffling.

Reply:
All set (used get_error_message).